sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncDatabase.java

```diff
@@ -533,6 +534,20 @@ public CosmosAsyncContainer getContainer(String id) {
         return withContext(context -> createUserInternal(userProperties, context));
     }
 
+    /**
+     * Creates a client encryption key after subscription the operation will be performed. The
+     * {@link Mono} upon successful completion will contain a single resource
+     * response with the created client encryption key. In case of failure the {@link Mono} will
+     * error.
+     *
+     * @param keyProperties the cosmos client encryption key properties
+     * @return an {@link Mono} containing the single resource response with the
+     * created cosmos client encryption key or an error.
+     */
+    Mono<CosmosClientEncryptionKeyResponse> createClientEncryptionKey(CosmosClientEncryptionKeyProperties keyProperties) {
```

Comment:
can this somehow go to the encryption sub-module instead of the core sdk?

discuss in the public api meeting.

Reply:
As discussed , P1 item. Thanks for raising this in meeting aswell